tests/lib/config-array-factory.js

```diff
@@ -780,32 +801,37 @@ describe("ConfigArrayFactory", () => {
-                    cwd: () => tempDir,
-                    files: {
-                        "node_modules/eslint-config-foo/index.js": "exports.env = { browser: true }",
-                        "node_modules/eslint-config-one/index.js": "module.exports = { extends: 'two', env: { browser: true } }",
-                        "node_modules/eslint-config-two/index.js": "module.exports = { env: { node: true } }",
-                        "node_modules/eslint-config-override/index.js": `
-                            module.exports = {
-                                rules: { regular: 1 },
-                                overrides: [
-                                    { files: '*.xxx', rules: { override: 1 } },
-                                    { files: '*.yyy', rules: { override: 2 } }
-                                ]
-                            }
-                        `,
-                        "node_modules/eslint-plugin-foo/index.js": "exports.configs = { bar: { env: { es6: true } } }",
-                        "node_modules/eslint-plugin-invalid-config/index.js": "exports.configs = { foo: {} }",
-                        "node_modules/eslint-plugin-error/index.js": "throw new Error('xxx error')",
-                        "base.js": "module.exports = { rules: { semi: [2, 'always'] } };"
-                    }
-                });
+            const { prepare, cleanup, getPath } = createCustomTeardown({
+                cwd: tempDir,
+                files: {
+                    "node_modules/eslint-config-foo/index.js": "exports.env = { browser: true }",
+                    "node_modules/eslint-config-one/index.js": "module.exports = { extends: 'two', env: { browser: true } }",
+                    "node_modules/eslint-config-two/index.js": "module.exports = { env: { node: true } }",
+                    "node_modules/eslint-config-override/index.js": `
+                        module.exports = {
+                            rules: { regular: 1 },
+                            overrides: [
+                                { files: '*.xxx', rules: { override: 1 } },
+                                { files: '*.yyy', rules: { override: 2 } }
+                            ]
+                        }
+                    `,
+                    "node_modules/eslint-plugin-foo/index.js": "exports.configs = { bar: { env: { es6: true } } }",
+                    "node_modules/eslint-plugin-invalid-config/index.js": "exports.configs = { foo: {} }",
+                    "node_modules/eslint-plugin-error/index.js": "throw new Error('xxx error')",
+                    "base.js": "module.exports = { rules: { semi: [2, 'always'] } };"
+                }
+            });
 
-                factory = new ConfigArrayFactory();
+            factory = new ConfigArrayFactory({
+                cwd: getPath(),
+                eslintAllPath,
+                eslintRecommendedPath
             });
```

Comment:
Mocha runs all functions passed to all `describe`s right away, before running anything else. This should be in `before(() => { ... })` to execute at the right moment (should fix the first two failing tests).

```js
// logs 1 2 3 4

describe("", () => {
    console.log("1");
    it("", () => { 
        console.log("3"); 
    });
});

describe("", () => {
    console.log("2");
    it("", () => { 
        console.log("4"); 
    });
});
```

Reply:
Ooh thanks!